frontend/src/metabase/query_builder/components/filters/modals/BulkFilterModal/BulkFilterFieldSearch.tsx

```diff
@@ -10,23 +10,16 @@ import {
 export const FieldSearch = ({
   value,
   onChange,
+  isExpanded,
+  setIsExpanded,
 }: {
   value: string;
   onChange: (value: string) => void;
+  isExpanded: boolean;
+  setIsExpanded: (isExpanded: boolean) => void;
 }): JSX.Element => {
-  const [showSearch, setShowSearch] = useState(false);
   const inputRef = useRef<HTMLInputElement | null>(null);
 
-  useEffect(() => {
```

Comment:
I moved this logic up to the `BulkFilterModal` for a few reasons:
* Allows the FocusTrap to focus on the modal container, rather than the search box. This allows the user to navigate with tabs and prevents random input into the search box before it's opened.
* Also allows us to scope the event listener to the modal using `onKeyDown`, which I think is both more intuitive and cleaner in code
* It seemed like the event listener only seemed to trigger when the user first clicked on the modal (with the focus trap enabled), which doesn't make sense if we want to navigate only with the keyboard.